### ISSUE TITLE
websocket upgrade?

### ISSUE THREAD

**Issue by c-cube** [ygrek/ocurl#75]
I see in [curl's doc](https://curl.se/docs/websocket.html) that websockets are supported, but I don't see them in https://ygrek.org/p/ocurl/api/Curl.html . Is it explicitly not supported? Would it be a lot of work to add support, if there was a will?

**Comment by nojb** [comment]
I allow myself to give a first response (but of course @ygrek will give the ultimate answer). This project is basically on maintenance mode but patches to extend coverage of libcurl are received with some frequency; they are reviewed and generally accepted (as a glance at the list of closed PRs will show you). So I would say that if you are interested in adding bindings for the WebSocket functions, your best bet is to put a patch together and submit it for integration.

One issue to keep in mind is that these functions are marked "EXPERIMENTAL" which means they could change in the future. I don't know if there is any policy regarding such functions in `ocurl`.

**Comment by ygrek** [comment]
Yes, patches are welcome ofcourse. Experimental can just mark the same in ocaml interface and we will be free to break them in the future.
